src/MallardMessageHandlers.Tests/AuthenticationTokenHandlerTests.cs

```diff
@@ -31,7 +32,7 @@ Task SessionExpired(CancellationToken ct, HttpRequestMessage request, TestToken
 				=> Task.CompletedTask;
 
 			void BuildServices(IServiceCollection s) => s
-				.AddSingleton<IAuthenticationTokenProvider<TestToken>>(new AuthenticationTokenProvider<TestToken>(GetToken, SessionExpired))
+				.AddSingleton<IAuthenticationTokenProvider<TestToken>>(sp => new ConcurrentAuthenticationTokenProvider<TestToken>(sp.GetService<ILoggerFactory>(), GetToken, SessionExpired))
```

Comment:
Since we've moved most of the concurrent logic from the handler into the new token provider, I wondering if we should also move/rework the tests as well, so it's easier to maintain. Basically, create tests for the concurrentAuthenticationTokenProvider which will include most of the concurrent tests.

Reply:
I think this can be done in another PR. I'll take a note of it though.